onset_maproom/calc.py

```diff
@@ -13,6 +13,52 @@ def read_zarr_data(zarr_path):
 # Growing season functions
 
 
+def estimate_sm(
+    daily_rain,
+    et,
+    taw,
+    sminit,
+    time_coord="T",
+):
+    """Estimates soil moisture from
+    rainfall,
+    evapotranspiration,
+    total available water and
+    intial soil moisture value, knowing that:
+    sm(t) = sm(t-1) + rain(t) - et(t)
+    with roof and floor respectively at taw and 0 at each time step.
+    """
+    # Creating a soil_moisture array
+    soil_moisture = daily_rain * 0
+    # Intializing sm
+    soil_moisture = xr.where(
+        (soil_moisture.coords[time_coord] == soil_moisture.coords[time_coord][0]),
+        (
+            sminit
+            + daily_rain.sel(**{time_coord: daily_rain.coords[time_coord][0]})
+            - et
```

Comment:
In your test case `et` is a scalar, so this works, but the comment says that we expect it to be an array with dimension `t`, in which case shouldn't this be something like `et.sel(daily_rain.coords[time_coord][0])`?

Reply:
et should be same structure than daily_rain/soil_moisture. It's a scalar in Eth Maproom case. Shall I try cover the other cases?